SCRAM/GMake/Makefile.rules

```diff
@@ -1136,14 +1136,15 @@ $(eval $(call LogFile,$(1),$(2)))
 .PHONY: $(1) all_$(1)
 $(1)_objdir := $(WORKINGDIR)/$(2)/$(1)
 $(1)_srcdir       := $(LOCALTOP)/$(2)
-ifneq ($(strip $(CUDASRC_FILES_SUFFIXES)),)
 $(1)_cudafiles    := $$(filter $(foreach e,$(CUDASRC_FILES_SUFFIXES),%.$e),$($(1)_files))
+ifneq ($(strip $$($(1)_cudafiles)),)
 $(1)_cudaobj      := $$($(1)_objdir)/$(1)_scramcuda.$(OBJEXT)
 endif
-$(1)_files        := $$(filter-out $($(1)_SKIP_FILES) $$($(1)_cudafiles), $($(1)_files))
-$(1)_prodsrc      := $$(addprefix $(patsubst src/%,%,$2),$$($(1)_files) $($(1)_cudafiles))
-$(1)_files_exts   := $$(sort $$(patsubst .%,%,$$(suffix $$($(1)_files) $$($(1)_cudafiles))))
-$(1)_LOC_USE      += $$(if $$(strip $$(filter $(FORTRANSRC_FILES_SUFFIXES),$$($(1)_files_exts))),$(F77_TYPE_COMPILER),) $(if $(strip $(CUDASRC_FILES_SUFFIXES)),$$(if $$(strip $$(filter $(CUDASRC_FILES_SUFFIXES),$$($(1)_files_exts))),$(CUDA_TYPE_COMPILER),))
+$(1)_files        := $$(filter-out $($(1)_SKIP_FILES), $($(1)_files))
+$(1)_prodsrc      := $$(addprefix $(patsubst src/%,%,$2),$$($(1)_files))
+$(1)_files_exts   := $$(sort $$(patsubst .%,%,$$(suffix $$($(1)_files))))
+$(1)_LOC_USE      += $$(if $$(strip $$(filter $(FORTRANSRC_FILES_SUFFIXES),$$($(1)_files_exts))),$(F77_TYPE_COMPILER),)
+$(1)_LOC_USE      += $$(if $$(strip $$(filter $(CUDASRC_FILES_SUFFIXES),$$($(1)_files_exts))),$(CUDA_TYPE_COMPILER),)
```

Comment:
I split this in two and made the CUDA case similar to the fortran case - but I do not know what this does...

Reply:
this adds automatic dependency on
- fortran (if we have .f, .f77 source files)
- cuda if we have .cu files